### ISSUE TITLE
Event definitions stop working after server is crashed

### ISSUE THREAD

**Issue by frakti** [Graylog2/graylog2-server#8612]
## Current Behavior

After server is crashed (due to OOM) and started again we stopped getting notifications. 
We've noticed there is no any record in `Alerts & Events`. After digging further we could observe that Event Definitions were left in wrong state after the crash.
On screenshot below (taken at 2020-07-22) below you can see broken **HTTP 500** event definition left in running state with old execution times (2020-07-10). Problem touched all event definitions. 
Second event definition **At least on ERROR message** looks fine since we've disabled and enabled it, which fixed the state.

<img width="1195" alt="image" src="https://user-images.githubusercontent.com/679817/88314116-bf417700-cd14-11ea-8b5d-efa35a5e1510.png">

What's more, we've checked some Graylog metrics we collect via Prometheus and found that it touched all scheduled jobs:

<img width="1577" alt="image" src="https://user-images.githubusercontent.com/679817/88314567-67574000-cd15-11ea-9121-674fc738aa75.png">

So I suppose after our Graylog pod restart due to OOM job execution engine couldn't start properly. Unfortunately we didn't collect server logs from Graylog.
## Environment

* Graylog Version:  Graylog 3.3.2+ade4779
* MongoDB Version: 4.1.13
* Environment: kubernetes

**Comment by pawelrychlik** [comment]
Bump.

OOM in Graylog server leaves Event Definitions in a broken state. Event Definitions that happened to be running at the time when the OOM occurred, after server restart are left in a state "running" forever, and the next scheduled executions aren't picked up properly.

Graylog Version: 3.3.4+279ef2c

**Comment by pawelrychlik** [comment]
Bump. This problem has been bothering us for half a year now. Is there any known fix or workaround for this? (maybe apart from ensuring enough RAM for Graylog as to prevent OOMs 🙃; as a background info - we have two dozens of Graylog instances, they receive traffic that varies over time, i.e. even with sophisticated memory mgmt in kubernetes it's hard to prevent OOMs entirely).

For now, we resorted to implementing a shameful cronjob to cycle through the event definitions in Graylog (through it's API) and double check if they got stuck or not, and re-enable if needed. That still sucks.

**Comment by bjoernhaeuser** [comment]
@pawelrychlik could you share the cronjob or bash script you have? We are facing the same problem. Thanks

**Comment by pawelrychlik** [comment]
> @pawelrychlik could you share the cronjob or bash script you have? We are facing the same problem. Thanks

@bjoernhaeuser Truth is, the k8s+node.js cronjob that we've came up with... doesn't really work reliably. It runs every 5 mins, and uses Graylog API to:
1. list event definitions,
2. it verifies that they are in a `"runnable"` status, and that `triggered_at` isn't too much outdated (which is the case when this bug occurs),
3. and if so - it disables the event definition and enables it back (this is the "manual fix" that works in Graylog UI, just ported over to an automated script).
4. just to double check - we run through the event definitions again and see if they are runnable after the disable-enable trick. This sounds a bit stupid, but in fact it's not uncommon that they are still not runnable... 🤷‍♂️ 

Bottom line - as much as I'd like to help out, I'm afraid it's just not worth sharing code that doesn't really solve the problem.

PS. We are now digging deeper into Graylog, elasticsearch and fluentd 💤 🙄 🤞

**Comment by AntonEbel** [comment]
Hi @frakti, thank you for reporting this issue. The current behavior of graylog is that only the node on which the job was running and not shut down properly can resolve the hanging job on the next startup. For each Graylog node, a node ID is generated at first startup and stored in a file on the file system. The location of this file is configured in greylog.conf. I assume that in your scenario the old container is discarded and a new one is started with a new node ID. So the hanging job is never resolved from the old node. We work on a solution.
